### PR TITLE
4176 passive listeners

### LINKED ISSUE #4176: Optimization: Passive listeners and scrolling performance
### Background
Looking to [optimize the site's performance](https://github.com/fecgov/fec-cms/issues/3711) and that of the homepage specifically, the Lighthouse tool in Chromium browsers has made some suggestions. Some of these will have repercussions throughout the app so we're going to make them their own tickets.

# Recommendation: Does not use passive listeners to improve scrolling performance
Consider marking your touch and wheel event listeners as `passive` to improve your page's scroll performance. [details](https://web.dev/uses-passive-event-listeners/?utm_source=lighthouse&utm_medium=devtools)

## Overview
We could add `, {passive: true}` to the scroll listeners.

## Benefit (the 80 of 80/20)
It's recommended but not sure know who might notice

## Effort (the 20 of 80/20)
Minimal

## Complications
- Being that these are third-party code (e.g. recaptcha), we may not be able to do this (like if the listeners are automatic and the code doesn't give us an option for passive)
- Would just need to check that recaptcha and other scroll-listeners still work

#### Related tickets
- [Preload key requests](https://github.com/fecgov/fec-cms/issues/4168)
- [Remove unused JavaScript](https://github.com/fecgov/fec-cms/issues/4169)
- [Remove unused CSS](https://github.com/fecgov/fec-cms/issues/4170)
- [Preconnect to required origins](https://github.com/fecgov/fec-cms/issues/4171)
- [Eliminate render-blocking resources](https://github.com/fecgov/fec-cms/issues/4172)
- [Ensure text remains visible during webfont load](https://github.com/fecgov/fec-cms/issues/4173)
- [Reduce the impact of third-party code](https://github.com/fecgov/fec-cms/issues/4175)
- [Image elements do not have explicit width and height](https://github.com/fecgov/fec-cms/issues/4177)
- [Serve static assets with an efficient cache policy](https://github.com/fecgov/fec-cms/issues/4178)
- [Minimize main-thread work](https://github.com/fecgov/fec-cms/issues/4179)
- [Avoid an excessive DOM size](https://github.com/fecgov/fec-cms/issues/4180)
- [Reduce JavaScript execution time](https://github.com/fecgov/fec-cms/issues/4181)
- [Avoid enormous network payloads](https://github.com/fecgov/fec-cms/issues/4182)
- [[aria-hidden="true"] elements contain focusable descendents](https://github.com/fecgov/fec-cms/issues/4183)
- [Elements with an ARIA [role] that require children to contain a specific [role] are missing some or all of those required children](https://github.com/fecgov/fec-cms/issues/4184)
- [[aria-*] attributes do not have valid values](https://github.com/fecgov/fec-cms/issues/4185)
- [ARIA IDs are not unique](https://github.com/fecgov/fec-cms/issues/4186)
- [Lists do not contain only <li> elements and script supporting elements (<script> and <template>)](https://github.com/fecgov/fec-cms/issues/4187)
- [[user-scalable="no"] is used in the <meta name="viewport"> element or the [maximum-scale] attribute is less than 5](https://github.com/fecgov/fec-cms/issues/4188)
- [Links to cross-origin destinations are unsafe](https://github.com/fecgov/fec-cms/issues/4189)
- [Serves images with low resolution](https://github.com/fecgov/fec-cms/issues/4190)

### PR DESCRIPTION
**Pull request by rfultz**
## Summary

- Resolves #4176 

Converting listeners to `passive` (when/where possible) to improve page performance. As far as our performance goes, passive listeners only matter when looking at touchstart and scroll events.

### Required reviewers

UX and/or front-end devs

## Impacted areas of the application

Because passive listeners are a breaking change for [browsers that don't support them](https://caniuse.com/passive-event-listener), implementing them is less straight-forward that one would expect.

We're getting flagged for non-passive listeners coming from our vendor file, recaptcha, and contributions-by-state.
- vendor
  - Updated leaflet, hoping that will alleviate the warning
  - An upcoming [major version of jQuery](https://github.com/jquery/jquery/issues/2871) will support passive listeners but it's a big change and unwieldy with the current version. jQuery 4.0 has no expected release date; it's [currently estimated at 73% complete](https://github.com/jquery/jquery/milestone/7)
- Recaptcha is a work in progress—looking into options
- Where Contributions Come From has been updated to use passive listeners
- The presidential finance map has been updated to use passive listeners

## Screenshots

Should be no visible changes

## How to test

This should not change anything in most modern browsers, apart from improve performance while scrolling, particularly for touch devices.

Because they could break older browsers, we need to test in Internet Explorer, Chrome ≤ 50, Firefox ≤ 48, Safari < 10 (OS X and iOS), Edge ≤ 15

- pull the branch
- `npm i`
- `npm run build`
- `./manage.py runserver`
- Make sure the JavaScript file for WCCF isn't included in the Lighthouse complaint about passive listeners
- Make sure the [presidential finance map](http://127.0.0.1:8000/data/candidates/president/presidential-map/) doesn't have its own entry in the Lighthouse passive listeners alert. **Also make sure the 'export raising' window scroll functionality works as expected**

## System architecture updates

None

### CODE REVIEW

**Comment by JonellaCulmer** [comment]
@rfultz I don't have any of those browser versions to test. Recommendations?

**Review by johnnyporkchops — APPROVED** [review]
Tested Pres Map in IE 11 and IE-Edge  MS Surface . The the 'export raising' window scroll functionality works as expected for both mouse and touchscreen interaction

The Lighthouse passive listener alerts for  `contributions-by-state-box.js` are gone on `/data/raising-bythenumbers/`

File: fec/fec/static/js/widgets/pres-finance-map-box.js [hunk 1]
```diff
@@ -1257,49 +1257,39 @@ PresidentialFundsMap.prototype.openDownloads = function() {
...
+  document.addEventListener(
+    'scroll',
+    this.handleDocScrolling.bind(this),
+    passiveListener()
+  );
 };
 
 /**
- * TODO -
+ * Fired on each tick/update of the scroll position. Removes its own listener when scroll is complete
```
Comment (johnnyporkchops) [inline fec/fec/static/js/widgets/pres-finance-map-box.js:1272]: The open/close toggle does not seem to work when scrolling away and coming back to it. We could simplify this interaction if it solves this problem by, doe example, opening the it on-click and then not offering an option to toggle it open/close it once its open
